src/cpuprofiler.js

```diff
@@ -620,24 +623,23 @@ var emscriptenCpuProfiler = {
       if (typeof glCtx[f] !== 'function' || f.startsWith('real_')) continue;
       this.hookWebGLFunction(f, glCtx);
     }
-    var this_ = this;
     // The above injection won't work for texImage2D and texSubImage2D, which have multiple overloads.
-    glCtx['texImage2D'] = function(a1, a2, a3, a4, a5, a6, a7, a8, a9) { 
-      this_.enterSection(1);
+    glCtx['texImage2D'] = (a1, a2, a3, a4, a5, a6, a7, a8, a9) => {
+      this.enterSection(1);
       var ret = (a7 !== undefined) ? glCtx['real_texImage2D'](a1, a2, a3, a4, a5, a6, a7, a8, a9) : glCtx['real_texImage2D'](a1, a2, a3, a4, a5, a6);
-      this_.endSection(1);
+      this.endSection(1);
       return ret;
     };
-    glCtx['texSubImage2D'] = function(a1, a2, a3, a4, a5, a6, a7, a8, a9) { 
-      this_.enterSection(0);
+    glCtx['texSubImage2D'] = (a1, a2, a3, a4, a5, a6, a7, a8, a9) => {
+      this.enterSection(0);
       var ret = (a8 !== undefined) ? glCtx['real_texSubImage2D'](a1, a2, a3, a4, a5, a6, a7, a8, a9) : glCtx['real_texSubImage2D'](a1, a2, a3, a4, a5, a6, a7);
-      this_.endSection(0);
+      this.endSection(0);
       return ret;
     };
-    glCtx['texSubImage3D'] = function(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11) { 
-      this_.enterSection(0);
+    glCtx['texSubImage3D'] = (a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11) => {
+      this.enterSection(0);
       var ret = (a9 !== undefined) ? glCtx['real_texSubImage3D'](a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11) : glCtx['real_texSubImage2D'](a1, a2, a3, a4, a5, a6, a7, a8);
-      this_.endSection(0);
+      this.endSection(0);
```

Comment:
The changes from `this_` to `this` will cause a bug. Previously, `this_` referred to the CpuProfiler` object, but after this change, `this` will reference `glCtx`.

Reply:
I thought the difference with arrow functions was that they don't get their own `this` so this kind of `var this = that` is no longer necessary.. the arrow function instead binds the this from its outer scope, right?

Reply:
From the spec: https://262.ecma-international.org/6.0/#sec-arrow-function-definitions-runtime-semantics-evaluation

```
An ArrowFunction does not define local bindings for arguments, super, this, or new.target. Any reference to arguments, super, this, or new.target within an ArrowFunction must resolve to a binding in a lexically enclosing environment. Typically this will be the Function Environment of an immediately enclosing function.
```

So I think the `that = this` trick is no needed if we use arrow functions.